server/rest_slicer_cli.py

```diff
@@ -973,59 +972,57 @@ def genRESTEndPointsForSlicerCLIsInDockerCache(info, restResource, dockerCache):
     -------
 
     """
+
     dockerImages = dockerCache.getDockerImageList()
     # validate restResource argument
     if not isinstance(restResource, (str, Resource)):
-        raise Exception('restResource must either be a string or '
-                        'an object of girder.api.rest.Resource')
-
-    # create REST resource if given a name
-    if isinstance(restResource, str):
-        restResource = type(restResource,
-                            (Resource, ),
-                            {'resourceName': restResource})()
-
-    restResourceName = type(restResource).__name__
-
-    # Add REST routes for slicer CLIs in each docker image
-    cliList = []
+        raise Exception('restResource must either be a '
```

Comment:
The resource has to be a Docker Resource only

if not isinstance(restResource, DockerResource):
        raise Exception('restResource must be a '
                        'Docker Resource')